src/configuration/settings/s3.cr

```diff
@@ -0,0 +1,16 @@
+class Configuration::Settings::S3
+  getter errors : Array(String)
+  getter s3 : Configuration::S3
+
+  def initialize(@errors, @s3)
+  end
+
+  def validate
+    case s3.enabled
```

Comment:
Perhaps you could simplify this a bit since you are only checking one condition and the `case` statement is overkill.

```suggestion
    return unless s3.enabled
    
    errors << "access_key is required for S3 backups" if s3.access_key.strip.empty?
    errors << "secret_key is required for S3 backups" if s3.secret_key.strip.empty?
    errors << "bucket is required for S3 backups" if s3.bucket.strip.empty?
```